bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/patch/EMFCommandHandler.java

```diff
@@ -71,7 +71,7 @@ private <C> PatchCommand<CCommand> getPatchCommand(final C ctx, final JsonNode p
 
    @FunctionalInterface
    private interface Decoder<C> {
-      Optional<EObject> decode(C context, String payload) throws DecodingException;
+      Optional<EObject> decode(String modelUri, C context, String payload) throws DecodingException;
```

Comment:
I think a doc comment to indicate that the `modelUri` can be `null` would be helpful. It isn't at all obvious that it would be valid to request a decode for a `null` model URI.

Reply:
Perhaps make it even more obvious like that?
```suggestion
      Optional<EObject> decode(@Nullable String modelUri, C context, String payload) throws DecodingException;
```

Reply:
That could be cool, yes! I wasn't sure whether we are using these annotations in this project. I think at least that they are not consistently applied.

Reply:
Yes I agree, until now they are rarely used, we should probably pursue this further from now on :)

Reply:
I checked this quickly, we use this in 4 classes in 10 places, so definitely not that often.
Also which annotation to use, there is no standard `javax.annotation` for this. In the ten cases we use the jetbrain one, but there is also an eclipse one `org.eclipse.jdt.annotation.NonNull` which can be use in eclipse null analysis which we did not enable.
I would suggest to create a ticket to discuss this and enable null analysis and unify the annotations. Do you agree?